lib/Translation/ExportVerilog/ExportVerilogFIRRTL.cpp

```diff
@@ -1852,10 +1859,11 @@ void ModuleEmitter::collectNamesEmitDecls(Block &block) {
 
   // Okay, now that we have measured the things to emit, emit the things.
   for (auto value : valuesToEmit) {
-    ops.clear();
```

Comment:
You're right. This will have some unintended consequences (which weren't being checked in the tests). I'll stop overusing this and switch to a separate set to track declarations across iterations of this loop. Will be fixed in a fixup commit.